transpiler/test/evm.cpp

```diff
@@ -0,0 +1,1632 @@
+
+        typename placeholder_public_preprocessor<field_type, kzg_placeholder_params_type>::preprocessed_data_type
+            kzg_preprocessed_public_data =
+                placeholder_public_preprocessor<field_type, kzg_placeholder_params_type>::process(
+                    constraint_system,
+                    assignments.public_table(),
+                    desc,
+                    kzg_scheme
+                );
+
+        typename placeholder_private_preprocessor<field_type, kzg_placeholder_params_type>::preprocessed_data_type
+            kzg_preprocessed_private_data =
+                placeholder_private_preprocessor<field_type, kzg_placeholder_params_type>::process(
+                    constraint_system,
+                    assignments.private_table(),
+                    desc
+                );
+
+        std::string output = std::string("circuit_kzg_") + curve_name<curve_type>();
+
+        auto printer = nil::blueprint::kzg_evm_verifier_printer<kzg_placeholder_params_type>(
+            constraint_system,
+            kzg_preprocessed_public_data.common_data,
+            output,
+            26, // gates library size threshold
+            60, // lookups library size threshold
+            13, // gates inline size threshold
+            15  // lookups inline size threshold
+        );
+        printer.print();
+
+        if (should_save_proof_data()) {
+            auto kzg_proof = placeholder_prover<field_type, kzg_placeholder_params_type>::process(
+                kzg_preprocessed_public_data,
+                std::move(kzg_preprocessed_private_data),
+                desc,
+                constraint_system,
+                kzg_scheme
+            );
+
+            using common_data_type = typename placeholder_public_preprocessor<field_type, kzg_placeholder_params_type>::
+                preprocessed_data_type::common_data_type;
+
+            print_placeholder_proof_with_params<Endianness, kzg_placeholder_params_type>(
+                kzg_preprocessed_public_data,
+                kzg_proof,
+                kzg_scheme,
+                desc,
+                output
+            );
+            print_public_input(
+                desc.public_input_columns == 0 ? std::vector<typename field_type::value_type>({})
+                                               : assignments.public_input(0),
+                output + "/" + public_input_filename
+            );
+
+            bool verifier_res = placeholder_verifier<field_type, kzg_placeholder_params_type>::process(
+                kzg_preprocessed_public_data.common_data,
+                kzg_proof,
+                desc,
+                constraint_system,
+                kzg_scheme
+            );
+            BOOST_CHECK(verifier_res);
+        }
+
+        return true;
+    }
+
+    plonk_table_description<field_type> desc;
+};
+
+BOOST_AUTO_TEST_SUITE(placeholder_circuit2_kzg_v2)
+
+using TestFixtures = boost::mpl::list<
```

Comment:
Please, add nil::crypto3::algebra::curves::alt_bn128_254::base_field_type curve here.

Reply:
Oh, right, it's the one with built-in support in EVM. Done.